### ISSUE TITLE
Preference Settings Window

### ISSUE THREAD

**Issue by marc-antoine-girard** [Thundernerd/Unity3D-SerializableInterface#48]
**Describe the solution you'd like**
This is mostly related to visual stuff. It would be nice to have options to change how SerializableInterface is displayed using the **Preference Window**. This would make visual decisions easier, as it would give us the ability to customize them. Ex:
- The compact view added in this PR: #42. While I like it, some might not, so it would be nice to have an option in Preference to enable or disable the compact mode.
- I tried a while ago to do something that would enable us to decide how we want to format the name. This allowed me to always see the interfaces. (While it was nice to see I had an instance of a `Dog`, it was also really helpful to see what interface I cared about in the `Dog` instance).
![image](https://user-images.githubusercontent.com/62125329/181272583-22fc53af-9d4f-4ccd-a73d-08bcb1a6f2aa.png)
![image](https://user-images.githubusercontent.com/62125329/181272686-750573ec-84b8-4b88-a4c2-90ebcaf9f810.png)
- Prefabs are added to the Assets locator. While this is a nice feature for Editor Scripting, it might not hold any value to some and would only make ScriptableObject harder to find. Adding a Preference to disable the Prefab finder would be a personal choice and wouldn't block users to simply drag and drop the prefab in the objectfield. (this is the default behaviour of Unity)

For now, the things I would see in the preference settings are:
- **Customize name for:** 
    - **Custom Class**
        - Default Display
        - Custom with variables: {Class} {interface} 
    - **Scene Object**
        - Default Display
        - Custom with variables: {GameObject} {Component} {interface}
    - **Assets**
        - Default Display
        - Custom with variables: {Asset} {interface}
- Enable / Disable Prefab discovery
- Enable / Disable Compact View
    - Color of Line if enabled 
- Reset / default button

**Additional context**
https://docs.unity3d.com/ScriptReference/ScriptableSingleton_1.html
https://docs.unity3d.com/ScriptReference/FilePathAttribute.html
https://docs.unity3d.com/ScriptReference/SettingsProvider.html

**Comment by Thundernerd** [comment]
This could be useful yep. I will look into this soon, hopefully, and might be able to bring back the vertical line drawing through this as well.

**Comment by marc-antoine-girard** [comment]
@Thundernerd Yes I saw you removed the line. No harm done. We transformed the line into a background color instead on our end since most of my team preferred it like that. Which is why the preference window would be really helpful. I prefer the line, they prefer the background, you prefer the Unity native look. That's a dropdown right there in the Preference window, and not so hard to implement.

**Comment by lordubbe** [comment]
It would be really nice to extend this settings menu with the ability to choose which types of references are included in the reference picker by default. Opening the dropdown is painfully slow (like, we're talking multiple seconds)

**Comment by lgarczyn** [comment]
> It would be really nice to extend this settings menu with the ability to choose which types of references are included in the reference picker by default. Opening the dropdown is painfully slow in larger projects (like, we're talking multiple seconds).

As a way to make it faster, it would be pretty easy to use a postprocessor to build an interface cache

Alternatively, caching the previous result and pre-displaying the dropdown while waiting for the rest of assets would also make it much faster
